lib/datadog/core/remote/client.rb

```diff
@@ -145,6 +161,57 @@ def payload
             cached_target_files: state.cached_target_files,
           }
         end
+
+        def ruby_engine_version
+          @ruby_engine_version ||= defined?(RUBY_ENGINE_VERSION) ? RUBY_ENGINE_VERSION : RUBY_VERSION
+        end
+
+        def gem_spec(name)
+          (@gem_specs ||= {})[name] ||= ::Gem.loaded_specs[name] || GemSpecificationFallback.new(nil, nil)
+        end
+
+        def native_platform
+          return @native_platform unless @native_platform.nil?
+
+          os = if RUBY_ENGINE == 'jruby'
+                 os_name = java.lang.System.get_property('os.name')
+
+                 case os_name
+                 when /linux/i then 'linux'
+                 when /mac/i   then 'darwin'
+                 else raise PlatformError, "unsupported JRuby os.name: #{os_name.inspect}"
+                 end
+               else
+                 Gem::Platform.local.os
+               end
+
+          version = if os != 'linux'
+                      nil
+                    elsif RUBY_PLATFORM =~ /linux-(.+)$/
+                      # Old rubygems don't handle non-gnu linux correctly
+                      Regexp.last_match(1)
+                    else
+                      'gnu'
+                    end
+
+          cpu = if RUBY_ENGINE == 'jruby'
+                  os_arch = java.lang.System.get_property('os.arch')
+
+                  case os_arch
+                  when 'amd64' then 'x86_64'
+                  when 'aarch64' then os == 'darwin' ? 'arm64' : 'aarch64'
+                  else raise PlatformError, "unsupported JRuby os.arch: #{os_arch.inspect}"
```

Comment:
We raise `PlatformError` and never rescue it. I'm worried this could lead to endless retries since the worker loop rescue `StandardError` and reset the client state. But we will end up with the same problem over and over.

Should we rescue it? 

Also, I see the test do not exercise the case where something fails, we should account for that as well in the tests

Reply:
Replaced by returning the original value.